nuxeo-runtime/nuxeo-connect-standalone/src/test/java/org/nuxeo/launcher/connect/TestConnectBroker.java

```diff
@@ -258,6 +260,36 @@ public void testInstallPackageRequest() throws Exception {
         checkLogEvents(expectedLogs, logCaptureResult.getCaughtEvents());
     }
 
+    /**
+     * Downloading a remote package fails if such package does not exist in remote. An explicit error message is shown:
+     * <pre>
+     * Download failed (not found).
+     * </pre>
+     */
+    @Test
+    @LogCaptureFeature.FilterWith(PkgRequestLogFilter.class)
+    // NXP-21573
+    public void testDownloadUnknownPackage() throws Exception {
+        // Environment setup
+        Environment.getDefault().setProperty(Environment.DISTRIBUTION_NAME, "server");
+        Environment.getDefault().setProperty(Environment.DISTRIBUTION_VERSION, "8.3");
+        ConnectBroker connectBroker = new ConnectBroker(Environment.getDefault());
+        ((StandaloneCallbackHolder) NuxeoConnectClient.getCallBackHolder()).setTestMode(true);
```

Comment:
I thought about it. But this problem is that it's not related to my changes... Should I still refactor it? Actually, I've noted other improvements about this class, too. Maybe I'm allowed to handle them all, too?

- [ ] Use annotation on top of each test for environment configuration
- [ ] Reconsider how to fix the assertions for log
- [ ] Fix attribute visibility
- [ ] Variable / method renaming: Use assert instead of check
- [ ] Fix spelling error
- [ ] Use temporary file instead of `new File( ... )` in the test